build: update required for java8 to 18 migration

kapil-agnihotri opened the pull request:
ING-3267

kapil-agnihotri committed:
build: update required for java8 to 18 migration

ING-3267

thorsten-reitz reviewed (approved):
(pretty much a pro forma review though, didn't test locally)

kapil-agnihotri committed:
build: update groovy to 2.5 instead of 4.0.1

Due to limitations of groovy-sandbox downgrading
groovy version to 2.5 instead of 4.0.1

ING-3267
